### PR TITLE
WalletConnect doesn't work for certain TX Signing, build 3.14(370) #2462

### LINKED ISSUE #2462: WaletConnect doesn't work for certain TX Signing, build 3.14(370)
The same TX doesn't work on iOS
IMG_1725

But works on Andriod 
photo_2021-01-19 16 33 33

### PR DESCRIPTION
**Pull request by vladyslav-iosdev**
Fixes #2462

### CODE REVIEW

**Comment by vladyslav-iosdev** [comment]
@hboon should be ok, tested with https://example.walletconnect.org/ worked for me fine.

**Comment by hboon** [comment]
Forced pushed to fix typo in commit message (important because might not be able to find "WalletConnect" otherwise)

**Comment by hboon** [comment]
From https://github.com/AlphaWallet/alpha-wallet-ios/issues/2462#issuecomment-762837760:

> and it appears because we cconnect to stored walletConnect sessions but we don't save urlToServer

But this only fixes the problem after the app is restarted, right?

I can't reproduce the issue with https://example.walletconnect.org/ if I don't restart the app even without applying the PR:

i.e

1. Don't apply PR
2. Connect
3. Don't restart app
4. Sign typed data
5. It works

**Comment by zhangzhongnan928** [comment]
> From [#2462 (comment)](https://github.com/AlphaWallet/alpha-wallet-ios/issues/2462#issuecomment-762837760):
> 
> > and it appears because we cconnect to stored walletConnect sessions but we don't save urlToServer
> 
> But this only fixes the problem after the app is restarted, right?
> 
> I can't reproduce the issue with https://example.walletconnect.org/ if I don't restart the app even without applying the PR:
> 
> i.e
> 
> 1. Don't apply PR
> 2. Connect
> 3. Don't restart app
> 4. Sign typed data
> 5. It works

Does the build you tested already support the new EIP712?
I just tested it again using the testflight build 3.14(370). I can reproduce it 
https://share.icloud.com/photos/0ViCFpAeE927zvs3KtlKM5dUw

**Comment by hboon** [comment]
> I just tested it again using the testflight build 3.14(370). I can reproduce it

I was wrong. I can reproduce it now.

File: AlphaWallet/InCoordinator.swift [hunk 1]
```diff
@@ -504,6 +504,7 @@ class InCoordinator: NSObject, Coordinator {
         coordinator.stop()
         removeAllCoordinators()
         OpenSea.resetInstances()
+        walletConnectCoordinator.disconnectAllSessions()
```
[the remark this answers is not in the document]
Comment (vladyslav-iosdev) [reply AlphaWallet/InCoordinator.swift:507]: we need to call disconnectAllSessions to notify connected WalletConnect application that we get disconnected. when change a wallet for example.

**Comment by vladyslav-iosdev** [comment]
> But this only fixes the problem after the app is restarted, right?

seems like yes, i was able to reproduce error only while switching between wallets, but its another issue.

**Comment by vladyslav-iosdev** [comment]
> I was wrong. I can reproduce it now.

@hboon can u tell me eexact steps to reproduce this? i followed these steps:

```
Don't apply PR
Connect
Don't restart app
Sign typed data
It works
```

**Comment by hboon** [comment]
>> I was wrong. I can reproduce it now. ...

@vladyslav-iosdev do you mean you cant reproduce this on master? It might have been fixed prior to this PR. Look for a commit around 8 jan.

**Comment by vladyslav-iosdev** [comment]
> > > I was wrong. I can reproduce it now. ...
> 
> @vladyslav-iosdev do you mean you cant reproduce this on master? It might have been fixed prior to this PR. Look for a commit around 8 jan.

i mean that i can't reproduce it with changes maded in #2469 (maybe i do something wrong)

**Comment by vladyslav-iosdev** [comment]
@hboon could this happens because instead of receiving `eth_signTypedData` we receive `eth_signTypedData_v3`?

**Comment by hboon** [comment]
Oh I think I confused you with my remark earlier. Let me try again. I believe this is true:

* I can now reproduce it if I checkout the commit around 8 Jan (I thought I couldn’t previously)
* I think it’s already fixed in master
* I can reproduce it after this PR is applied too

So if you can’t reproduce this anymore, it’s good :)

**Review by hboon — APPROVED** [review]
